io/module_write_netcdf.F90

```diff
@@ -358,10 +354,10 @@ subroutine write_netcdf(wrtfb, filename, &
                ncerr = nf90_def_var_chunking(ncid, varids(i), NF90_CHUNKED, chunksizes) ; NC_ERR_STOP(ncerr)
             end if
 
-            ishuffle = NF90_SHUFFLE
-            ! shuffle filter off for 3d fields using lossy compression
-            if (rank == 3 .and. nbits(grid_id) > 0) then
-                ishuffle = NF90_NOSHUFFLE
+            ishuffle = NF90_NOSHUFFLE
+            ! shuffle filter on when using lossy compression
+            if ( quantize_nsd(grid_id) > 0) then
+                ishuffle = NF90_SHUFFLE
```

Comment:
I want to confirm the code here. So previously there was no shuffle for lossy compression, there is shuffle for 2D fields. but now there is no shuffle for 2D fields and there is shuffle for 3D fields with lossy compression? Can we keep 2D fields shuffled?

Reply:
So basically shuffle should enabled whenever compression (deflate or zstandard) is enabled, is that correct?